### PR TITLE
Provide letter fallback for Avatar

### PR DESCRIPTION
**Pull request by danedavid**
* Avatar falls back to default letter if image does not load #156 
* Fix `Invalid prop` warning thrown inside Header component in docs

### CODE REVIEW

**Comment by christianvuerings** [comment]
Deploy preview for *gestalt* ready!

Built with commit ad337305bb1453a91f231e9233e12d3fc2358d92

https://deploy-preview-167--gestalt.netlify.com

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please add a [changelog.md](https://github.com/pinterest/gestalt/blob/master/CHANGELOG.md) entry for your changes.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: packages/gestalt/src/Avatar/Avatar.js [hunk 1]
```diff
@@ -50,6 +50,10 @@ const DefaultAvatar = ({ name }: { name: string }) => {
   );
 };
 
+type State = {
```
Comment (chrislloyd) [inline packages/gestalt/src/Avatar/Avatar.js]: Can this be an exact object? `{| isImageLoaded: boolean |}` and come below the `Props` type please?

File: packages/gestalt/src/Avatar/Avatar.js [hunk 2]
```diff
@@ -63,76 +67,87 @@ const sizes = {
...
-        >
+export default class Avatar extends React.PureComponent<AvatarProps, State> {
+  static propTypes = {
+    name: PropTypes.string.isRequired,
+    src: PropTypes.string,
+    size: PropTypes.oneOf(['sm', 'md', 'lg']),
+    verified: PropTypes.bool,
+  };
+
+  state: State = {
```
Comment (chrislloyd) [inline packages/gestalt/src/Avatar/Avatar.js]: I don't think you need the `: State` typecasting here.

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Bundle size increase
  </td>
    </tr>
  </tbody>
</table>



Filename | Size Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip
--- | --- | --- | --- | --- | --- | --- |
gestalt.js | :small_red_triangle: 0.4% | :small_red_triangle: 0.2% | 280.83 KB | 282.03 KB | 62.95 KB | 63.06 KB
gestalt.es.js | :small_red_triangle: 0.4% | :small_red_triangle: 0.2% | 278.2 KB | 279.39 KB | 62.58 KB | 62.68 KB
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by chrislloyd** [comment]
Thanks again @danedavid!

**Pull request merged**